pkg/cloudprovider/providers/vsphere/vclib/datastore.go

```diff
@@ -0,0 +1,92 @@
+package vclib
+
+import (
+	"github.com/golang/glog"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	"golang.org/x/net/context"
+)
+
+// Datastore extends the govmomi Datastore object
+type Datastore struct {
+	*object.Datastore
+	datacenter *Datacenter
+}
+
+// Disk defines interfaces for creating disk
+type Disk interface {
```

Comment:
Created new Object VirtualDisk and moved methods there
```
type VirtualDisk struct {
	DiskPath      string
	VolumeOptions VolumeOptions
	VMOptions     VMOptions
}
```

`func (disk *VirtualDisk) CreateVirtualDisk(ctx context.Context, datastore *Datastore) (err error) {`

`func (disk *VirtualDisk) DeleteVolume(ctx context.Context, datastore *Datastore) error {`